src/components/application_manager/test/commands/mobile/set_global_properties_request_test.cc

```diff
@@ -564,14 +604,179 @@ TEST_F(SetGlobalPropertiesRequestTest, Run_TTSIncorrectSyntax_Canceled) {
+  MessageSharedPtr msg = CreateMsgParams();
+  SmartObject menu_icon(smart_objects::SmartType_Array);
+  menu_icon[strings::value] = "meun_icon\\n";
+  (*msg)[strings::msg_params][strings::menu_icon] = menu_icon;
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(mock_message_helper_, VerifyImage(_, _, _))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+  EXPECT_CALL(mock_app_manager_, RemoveAppFromTTSGlobalPropertiesList(_))
+      .Times(0);
+
+  EmptyExpectationsSetupHelper();
+
+  SharedPtr<SetGlobalPropertiesRequest> command(
+      CreateCommand<SetGlobalPropertiesRequest>(msg));
+
+  ExpectInvalidData();
+
+  command->Run();
+}
+
+TEST_F(SetGlobalPropertiesRequestTest, Run_InvalidMenuTitle_INVALID_DATA) {
+  MessageSharedPtr msg = CreateMsgParams();
+  SmartObject menu_title(smart_objects::SmartType_Array);
+  menu_title = "menu_title\\n";
+  (*msg)[strings::msg_params][strings::menu_title] = menu_title;
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(mock_message_helper_, VerifyImageVrHelpItems(_, _, _)).Times(0);
+  EXPECT_CALL(mock_app_manager_, RemoveAppFromTTSGlobalPropertiesList(_))
+      .Times(0);
+
+  EmptyExpectationsSetupHelper();
+
+  SharedPtr<SetGlobalPropertiesRequest> command(
+      CreateCommand<SetGlobalPropertiesRequest>(msg));
+
+  ExpectInvalidData();
+
+  command->Run();
+}
+
+TEST_F(SetGlobalPropertiesRequestTest,
+       Run_InvalidLimitedCharacterList_INVALID_DATA) {
+  MessageSharedPtr msg = CreateMsgParams();
+  SmartObject limited_character_list(smart_objects::SmartType_Array);
+  limited_character_list[0] = "limited_character_list\\n";
+  (*msg)[strings::msg_params][strings::keyboard_properties]
+        [strings::limited_character_list] = limited_character_list;
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(mock_message_helper_, VerifyImageVrHelpItems(_, _, _)).Times(0);
+  EXPECT_CALL(mock_app_manager_, RemoveAppFromTTSGlobalPropertiesList(_))
+      .Times(0);
+
+  EmptyExpectationsSetupHelper();
+
+  SharedPtr<SetGlobalPropertiesRequest> command(
+      CreateCommand<SetGlobalPropertiesRequest>(msg));
+
+  ExpectInvalidData();
+
+  command->Run();
+}
+
+TEST_F(SetGlobalPropertiesRequestTest,
+       Run_InvalidAutoCompleteText_INVALID_DATA) {
+  MessageSharedPtr msg = CreateMsgParams();
+  SmartObject auto_complete_text(smart_objects::SmartType_Array);
+  auto_complete_text = "auto_complete_text\\n";
+  (*msg)[strings::msg_params][strings::keyboard_properties]
+        [strings::auto_complete_text] = auto_complete_text;
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(mock_message_helper_, VerifyImageVrHelpItems(_, _, _)).Times(0);
+  EXPECT_CALL(mock_app_manager_, RemoveAppFromTTSGlobalPropertiesList(_))
+      .Times(0);
```

Comment:
@okozlovlux These three expectations repeat in several tests and the only difference it `Times()` argument. Perhaps you can create a function with these arguments?